src/utils/propsAreEqual.ts

```diff
@@ -7,7 +7,13 @@ export function propsAreEqual<Props extends SketchProps = SketchProps>(
   previous: P5WrapperProps<Props>,
   next: P5WrapperProps<Props>
 ) {
-  const differences = diff(previous, next);
+  // set deepCompare to true if not specified by props
+  const deepCompare = next.deepCompare === undefined ? true : next.deepCompare;
 
-  return differences.length === 0;
+  if (deepCompare) {
+    const differences = diff(previous, next);
+    return differences.length === 0;
+  } else {
+    return undefined;
```

Comment:
There needs to still be diffing of some sort done here, otherwise every frame will trigger a re-render. I am also unsure if it is wise to pass a prop to control this behaviour, I think it should be done as part of a provider of some sort for example which allows hooking into the config. Perhaps a better option is to simply check the size of the props being passed in and above a certain limit, deciding which route is better (deep or shallow comparison). I am still unsure that this overall is a good idea though because in the past shallow comparisons caused a lot of re-rendering issues, we need to be really careful with this.

Reply:
Ok that mostly makes sense, but could you please elaborate on why passing a prop to control that behaviour would not be a good idea?

Reply:
Because the props are meant to be passed to the sketch, not control the generation of the sketches render, does that make sense?

Reply:
Yes, it does makes sense! I completely understand your concerns regarding both the role that`deepCompare` prop plays as well as the performance issues that my changes may impose. Nonetheless, I'm curious to see what you think about how my additions work in the  example I implemented (4th sketch in `app.tsx`). With the current changes in `propsAreEqual`, the sketch re-renders only when the props change so that seems ok to me. I also removed the `as any` considering that `propsAreEqual` should now only return a bool. However, If you're still not convinced that going down this path is not a good idea, I'm more than happy to explore different options. 👍

Reply:
Yeah, I see your point also, for now we can try it your way and see how it goes 🚀